common/src/main/java/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -48,6 +47,12 @@ typealias DiagnosticsCallback = Pair<(JSONObject) -> Unit, (PurchasesError, Bool
 /** @suppress */
 typealias ProductEntitlementCallback = Pair<(ProductEntitlementMapping) -> Unit, (PurchasesError) -> Unit>
 
+enum class PostReceiptErrorType {
+    CAN_BE_CONSUMED,
+    SERVER_ERROR,
+    CANNOT_BE_CONSUMED
+}
```

Comment:
Not sure if this is the best way, but basically, we can do 3 different things on a post receipt error. 
- Consume the purchase, even though it's an error (`CAN_BE_CONSUMED`)
- Enable offline entitlements without consuming (`SERVER_ERROR`)
- Nothing (`CANNOT_BE_CONSUMED`)
This is what I'm trying to represent here... Feedback on naming or alternatives are welcome.

Reply:
Yeah... I was trying to avoid having the logic of what to do in the backend class... But that's probably much simpler so will do that 👍

Reply:
Hmm the thing about it is that the behavior is currently exclusive... As in, we will never `CONSUME` if we are using offline entitlements. I think making it an array where we could potentially send both `CONSUME` and `USE_OFFLINE_ENTITLEMENTS` but we will not support both at the same time could be confusing...

I could rename `SHOULD_USE_OFFLINE_ENTITLEMENTS_AND_NOT_CONSUME` to `SHOULD_USE_OFFLINE_ENTITLEMENTS`. And assume the "not consumption" part to be explicit... What do you think?